include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -1233,6 +1265,28 @@ __brick_calc_mask_1(_ForwardIterator __first, _ForwardIterator __last, bool* __r
     return ::std::make_pair(__count_true, __size - __count_true);
 }
 
+template <class _DifferenceType, class _ForwardIterator, class _Bound, class _UnaryPredicate>
+std::pair<_DifferenceType, _ForwardIterator>
+__brick_calc_mask_1(_ForwardIterator __first, _ForwardIterator __last, _Bound __m, bool* __restrict __mask, _UnaryPredicate __pred,
+                    /*vector=*/::std::false_type) noexcept
+{
+    auto __count_true = _DifferenceType(0);
+    auto __size = __last - __first;
```

Comment:
Since `__first` and `__last` are forward iterators we should use `std::distance`, correct?